status.go

```diff
@@ -67,6 +67,59 @@ const (
 	versionError                  = "ERROR"
 )
 
+func (s *sequins) serveHealth(w http.ResponseWriter, r *http.Request) {
+	s.dbsLock.RLock()
+
+	status := status{DBs: make(map[string]dbStatus)}
+	for name, db := range s.dbs {
+		status.DBs[name] = copyDBStatus(db.status())
+	}
+
+	s.dbsLock.RUnlock()
+
+	ok := true
+	statuses := make(map[string]map[string]map[string]versionState)
+
+	// Iterate through all of the nodes of the local databases and ensure
+	// that the versions they own are all marked as available. A single
+	// version being in an unvailable state results in an error response.
+	for dbKey, dbValue := range status.DBs {
+		for versionKey, versionValue := range dbValue.Versions {
+			for nodeKey, nodeValue := range versionValue.Nodes {
+				_, exists := statuses[dbKey]
+				if !exists {
+					statuses[dbKey] = make(map[string]map[string]versionState)
+				}
+
+				_, exists = statuses[dbKey][versionKey]
+				if !exists {
+					statuses[dbKey][versionKey] = make(map[string]versionState)
+				}
```

Comment:
I'm not super thrilled about this map code, but I wasn't sure of a much better way to do it 😕